networkx/algorithms/isomorphism/vf2pp.py

```diff
@@ -166,18 +181,35 @@ def vf2pp_all_isomorphisms(G1, G2, node_labels=None, default_label=None):
     ------
     dict
         Isomorphic mapping between the nodes in `G1` and `G2`.
-
     """
     if G1.number_of_nodes() == 0 or G2.number_of_nodes() == 0:
         return False
 
+    # Create the degree dicts based on graph type
+    if G1.is_directed():
+        G1_degree = {
+            n: (in_degree, out_degree)
+            for (n, in_degree), (_, out_degree) in zip(G1.in_degree, G1.out_degree)
+        }
+        G2_degree = {
+            n: (in_degree, out_degree)
+            for (n, in_degree), (_, out_degree) in zip(G2.in_degree, G2.out_degree)
+        }
+    else:
+        G1_degree = dict(G1.degree)
+        G2_degree = dict(G2.degree)
+
     # Check that both graphs have the same number of nodes and degree sequence
-    if not nx.faster_could_be_isomorphic(G1, G2):
+    if G1.order() != G2.order():
+        return False
+    if sorted(d for _, d in G1_degree.items()) != sorted(
+        d for _, d in G2_degree.items()
+    ):
```

Comment:
```suggestion
    if sorted(G1_degree.values()) != sorted(G2_degree.values()):
```
Because `G1.degree` is replaced by `G1_degree` which is a dict we can use `values()` here. :)